test/unit/util/dependencies/poetry_dependencies_test.py

```diff
@@ -0,0 +1,122 @@
+import subprocess
+
+import pytest
+
+from exasol.toolbox.util.dependencies.poetry_dependencies import (
+    PoetryDependencies,
+    PoetryDependency,
+    PoetryGroup,
+    PoetryToml,
+)
+
+MAIN_GROUP = PoetryGroup(name="main", toml_section="project.dependencies")
+DEV_GROUP = PoetryGroup(name="dev", toml_section="tool.poetry.group.dev.dependencies")
+ANALYSIS_GROUP = PoetryGroup(
+    name="analysis", toml_section="tool.poetry.group.analysis.dependencies"
+)
+
+PYLINT = PoetryDependency(name="pylint", version="3.3.7", group=MAIN_GROUP)
+ISORT = PoetryDependency(name="isort", version="6.0.1", group=DEV_GROUP)
+BLACK = PoetryDependency(name="black", version="25.1.0", group=ANALYSIS_GROUP)
+
+DIRECT_DEPENDENCIES = {
+    MAIN_GROUP.name: [PYLINT],
+    DEV_GROUP.name: [ISORT],
+    ANALYSIS_GROUP.name: [BLACK],
+}
+
+
+@pytest.fixture(scope="module")
+def cwd(tmp_path_factory):
+    return tmp_path_factory.mktemp("test")
+
+
+@pytest.fixture(scope="module")
+def project_name():
+    return "project"
+
+
+@pytest.fixture(scope="module")
+def project_path(cwd, project_name):
+    return cwd / project_name
+
+
+@pytest.fixture(scope="module")
+def create_poetry_project(cwd, project_name, project_path):
```

Comment:
We could move this to be an  integration test, but given that we rely on certain behaviors in poetry, I think it's worthwhile to explicitly test that it does that. The previous tests were using saved text, which as we switch poetry versions will migrate without a clear way to see it.